### ISSUE TITLE
[ecmascript] InnaccurateNumericLiteral false positive with hex literals

### ISSUE THREAD

**Issue by cshu** [pmd/pmd#861]
**Affects PMD Version:** 6.0.0

**Rule:** [InnaccurateNumericLiteral](https://pmd.github.io/pmd-6.4.0/pmd_rules_ecmascript_errorprone.html#innaccuratenumericliteral)

**Description:** Hexadecimal literal gives warning like "The numeric literal '0x20' will have at different value at runtime."

**Code Sample demonstrating the issue:**

```
var x=0x20;
```

**Running PMD through:** CLI

**Comment by jsotuyod** [comment]
@cshu thanks for the report. The rule currently doesn't support any number base other than 10. Binary, hexa and octal (strictly not part of ES up to 8, and with the `0o` prefix) would trigger this same issue.

We need to extend the current `NumberLiteral` to understand which base the original number was expressed in.

**Comment by adangel** [comment]
The latest ES Spec: https://www.ecma-international.org/ecma-262/8.0/index.html

The Numeric Literals are specified here: https://www.ecma-international.org/ecma-262/8.0/index.html#sec-literals-numeric-literals

It seems, that Rhino only supports hex literals. Using octal (`0o` or `0O` prefix) or binary (`0b` or `0B` prefix) does not even result in a parsed number literal node. I've shortly tried the latest rhino version (1.7.9), but there seems to be no change.
